cms/db/filecacher.py

```diff
@@ -738,16 +771,17 @@ def put_file_from_fobj(self, src, desc=""):
 
             cache_file_path = os.path.join(self.file_dir, digest)
 
-            if not os.path.exists(cache_file_path):
-                os.rename(dst.name, cache_file_path)
-            else:
-                os.unlink(dst.name)
+            # Store the file in the backend. We do that even if the file
+            # was already in the cache (that is, we ignore the check above)
+            # because there's a (small) chance that the file got removed
+            # from the backend but somehow remained in the cache.
+            # We read from the temporary file before moving it to
+            # cache_file_path because the latter might be deleted before
+            # we get a chance to open it.
+            with open(dst.name, 'rb') as src:
```

Comment:
I'm not sure what you mean. We can't open it after renaming the file, can we? Or did you mean that we could reuse the file-like object `dst` directly without opening `dst.name` (even after renaming)?

Reply:
I meant that we're calling `save()` before renaming the file and while the file is private/temporary, so it could be opened in `save()` too, like before. But now with `save()` being merged this becomes irrelevant.